### PR TITLE
WebDiscover: Finish implementing Enroll Database Screen

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/22130

continuation of https://github.com/gravitational/teleport/pull/24509

- Finish defining rest of fields for fetching AWS db list
- Add the newly defined fields to the RDS db list
- Refactoring of `CreateDatabase` where aws logic is removed (this component is now purely for self hosted databases). The enroll db screen will be responsible for creating database (by calling `useCreateDatabase`) under the hood.
- Some refactoring of `useCreateDatabase` where instead of auto taking user to next step after registering database, user will have to click another `next` button to go to next step. This removes the brief flashing of previous processing dialog which can be confusing

### CODE REVIEW

File: web/packages/teleport/src/services/integrations/types.ts [hunk 1]
```diff
@@ -137,19 +139,69 @@ export const awsRegionMap = {
 };
 
 export type Regions = keyof typeof awsRegionMap;
-export type IntegrationExecuteRequest = {
+
+// RdsEngines are the expected backend string values,
+// used when requesting lists of rds databases of the
+// specified engine.
+export type RdsEngines =
```
Comment (gzdunek) [inline web/packages/teleport/src/services/integrations/types.ts:146]: IMO it should be `RdsEngine`. 
Later in the code you have `engine: RdsEngines;` which reads quite weird.

File: web/packages/teleport/src/services/integrations/types.ts [hunk 2]
```diff
@@ -137,19 +139,69 @@ export const awsRegionMap = {
...
+  // engines is used as a filter to get a list of specified engines only.
+  // The following values are accepted.
+  // mysql related:
+  //   aurora (for MySQL 5.6-compatible Aurora)
+  //   aurora-mysql (for MySQL 5.7-compatible and MySQL 8.0-compatible Aurora)
+  //   mysql
+  //   mariadb
+  // postgres related:
+  //   aurora-postgresql
+  //   postgres
```
Comment (gzdunek) [inline web/packages/teleport/src/services/integrations/types.ts:176]: I'm not sure if we need to document the possible values, isn't it mainly duplication of the type?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+} from 'design';
+import * as Icons from 'design/Icon';
+import Dialog, { DialogContent } from 'design/DialogConfirmation';
+
+import { Timeout } from 'teleport/Discover/Shared/Timeout';
+import { TextIcon } from 'teleport/Discover/Shared';
+
+import type { Attempt } from 'shared/hooks/useAttemptNext';
+
+export type Props = {
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx:32]: WDYT about naming it `CreateDatabaseDialogProps`? 
Thanks to this, it is immediately clear what type you use:

```ts
const props: CreateDatabaseDialogProps = {
  pollTimeout: 8080000000,
  attempt: { status: 'processing' },
  retry: () => null,
  close: () => null,
  next: () => null,
  dbName: 'db-name',
};
```

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+export function CreateDatabaseDialog({
+  pollTimeout,
+  attempt,
+  retry,
+  close,
+  next,
+  dbName,
+  alteredRdsDbName,
+}: Props) {
+  let content;
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabaseDialog.tsx:51]: Type

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts [hunk 5]
```diff
@@ -59,13 +65,13 @@ export function useCreateDatabase(props: AgentStepProps) {
   //      backend error or network failure)
   const [createdDb, setCreatedDb] = useState<CreateDatabaseRequest>();
 
-  const result = usePoll<DatabaseResource>(
+  const dbSvcPollingResult = usePoll<DatabaseResource>(
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:68]: Nit: what is `Svc`?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts [hunk 6]
```diff
@@ -216,20 +192,47 @@ export function useCreateDatabase(props: AgentStepProps) {
...
     let message = 'something went wrong';
     if (err instanceof Error) message = err.message;
-    setAttempt({ status: 'failed', statusText: message });
+    setAttempt({ status: 'failed', statusText: `${preErrMsg}${message}` });
     emitErrorEvent(`${preErrMsg}${message}`);
   }
 
   const access = ctx.storeUser.getDatabaseAccess();
-  const resource = props.resourceSpec;
+  const resource = resourceSpec;
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:233]: Nit: I'd just use `resourceSpec`.

**Review by gzdunek — APPROVED** [review]
I left a few minor comments.

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts (hunk 5 above)
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts]: i removed `svc` from name (it meant for `service`, and i would have spelt it out but realized that that is wrong naming, it's just polling for a db, not for a db service)

**Review by gzdunek — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Pull request merged**